Network/Bluetooth/DeviceInfo.cs

```diff
@@ -28,77 +30,104 @@
         /// </summary>
-        /// <value><c>true</c> if this instance is known; otherwise, <c>false</c>.</value>
-        public bool IsKnown { get; private set; }
+        public DateTime LastSeen { get; }
 
         /// <summary>
-        /// Gets the signal strength.
+        /// The last time that the device was used.
         /// </summary>
-        /// <value>The signal strength.</value>
-        public int SignalStrength { get; private set; }
+        public DateTime LastUsed { get; }
 
         /// <summary>
-        /// Gets the last seen.
+        /// The <see cref="InTheHand.Net.Sockets.BluetoothDeviceInfo"/> for the
+        /// device.
         /// </summary>
-        /// <value>The last seen.</value>
-        public DateTime LastSeen { get; private set; }
+        internal BluetoothDeviceInfo BluetoothDeviceInfo { get; }
+
+        #endregion Properties
+
+        #region Constructors
 
         /// <summary>
-        /// Gets the last used.
+        /// Constructs and returns a new instance of the <see cref="DeviceInfo"/>,
+        /// mapping the given <see cref="InTheHand.Net.Sockets.BluetoothDeviceInfo"/>
+        /// to the <see cref="DeviceInfo"/>.
         /// </summary>
-        /// <value>The last used.</value>
-        public DateTime LastUsed { get; private set; }
+        /// <param name="deviceInfo">
+        /// The <see cref="InTheHand.Net.Sockets.BluetoothDeviceInfo"/> whose properties
+        /// to use for the new <see cref="DeviceInfo"/> instance.
+        /// </param>
+        internal DeviceInfo(BluetoothDeviceInfo deviceInfo)
+        {
+            DeviceName = deviceInfo.DeviceName;
+            IsKnown = deviceInfo.Remembered;
+            SignalStrength = deviceInfo.Rssi;
+            LastSeen = deviceInfo.LastSeen;
+            LastUsed = deviceInfo.LastUsed;
+            BluetoothDeviceInfo = deviceInfo;
+        }
+
+        #endregion Constructors
+
+        #region Methods
 
         /// <summary>
-        /// Gets the bluetooth device information.
+        /// For each of the given <see cref="InTheHand.Net.Sockets.BluetoothDeviceInfo"/>s,
+        /// generates a <see cref="DeviceInfo"/> and returns the generated array.
         /// </summary>
-        /// <value>The bluetooth device information.</value>
-        internal BluetoothDeviceInfo BluetoothDeviceInfo { get; private set; }
+        /// <param name="infos">
+        /// An array of <see cref="InTheHand.Net.Sockets.BluetoothDeviceInfo"/>s,
+        /// for each of which to generate the corresponding <see cref="DeviceInfo"/>.
+        /// </param>
+        /// <returns>
+        /// An array of <see cref="DeviceInfo"/>s, one for each of the given
+        /// <see cref="InTheHand.Net.Sockets.BluetoothDeviceInfo"/>s.
+        /// </returns>
+        internal static DeviceInfo[] GenerateDeviceInfos(BluetoothDeviceInfo[] infos)
+        {
+            return infos.ToList().Select(info => new DeviceInfo(info)).ToArray();
```

Comment:
Not wrong or anything. Just so u know.
You may use expression bodies for such "short" loc.

The lib is inconsistent with expression bodies, since back in 2015 that feature was not available.
But I try to add them, if the loc is small, if I see one to replace.

Reply:
I dont like expression body. Im just a bit verbose that way. I conceede that sometimes it makes more sense but i like to delimit my methods clearly. I will look out for this and attempt to use expression body style methods more.